DEVELOPER.md

```diff
@@ -93,6 +93,46 @@ make build/kumactl
 ```
 This could help expedite your development process if you only made changes to the `kumactl` files.
 
+### Experimental caching
+
+#### Requirements
+- sha256sum (`brew install coreutils`, usually already installed on linux)
+
+Make target `images` always rebuilds the image even if nothing has changed.
+This can eat up some time which is really precious in the developer feedback loop.
+
+The caching mechanism uses sha256 checksum and an additional docker tag to check if the files used for docker context actually changed.
```

Comment:
I'm confused isn't that something docker does from the start?

Reply:
Yeah, doesn't `make images` always rebuild because it runs `go build` and copies the binary into the images? Can we just build our binaries in docker?

Reply:
The thing is that our binaries are never idempotent because of: 

https://github.com/kumahq/kuma/blob/0ec42aa981139b1552e695b17e0f474cd0d5bde9/mk/build.mk#L7-L12

I wanted to make so that these are set differently when not in CI (and rely on go cache for builds) but I didn't get around to do it yet.

Reply:
> I'm confused isn't that something docker does from the start?

I thought that docker cached only the resulting layers not the context sent to the daemon, but you're right. 

> Yeah, doesn't make images always rebuild because it runs go build and copies the binary into the images?

I'm guessing that the resulting binary is different, that's why docker doesn't cache it.

> Can we just build our binaries in docker?

We shouldn't, we don't need all of the build tools to be in our images.

> I wanted to make so that these are set differently when not in CI (and rely on go cache for builds) but I didn't get around to do it yet.

Yeah that would be the preferred option.